src/parser/parseTransformAttribute.ts

```diff
@@ -1,105 +1,50 @@
 
 // == begin transform regexp
-const number = reNum,
-  skewX = '(?:(skewX)\\s*\\(\\s*(' + number + ')\\s*\\))',
-  skewY = '(?:(skewY)\\s*\\(\\s*(' + number + ')\\s*\\))',
-  rotate =
-    '(?:(rotate)\\s*\\(\\s*(' +
-    number +
-    ')(?:' +
-    commaWsp +
-    '(' +
-    number +
-    ')' +
-    commaWsp +
-    '(' +
-    number +
-    '))?\\s*\\))',
-  scale =
-    '(?:(scale)\\s*\\(\\s*(' +
-    number +
-    ')(?:' +
-    commaWsp +
-    '(' +
-    number +
-    '))?\\s*\\))',
-  translate =
-    '(?:(translate)\\s*\\(\\s*(' +
-    number +
-    ')(?:' +
-    commaWsp +
-    '(' +
-    number +
-    '))?\\s*\\))',
-  matrix =
-    '(?:(matrix)\\s*\\(\\s*' +
-    '(' +
-    number +
-    ')' +
-    commaWsp +
-    '(' +
-    number +
-    ')' +
-    commaWsp +
-    '(' +
-    number +
-    ')' +
-    commaWsp +
-    '(' +
-    number +
-    ')' +
-    commaWsp +
-    '(' +
-    number +
-    ')' +
-    commaWsp +
-    '(' +
-    number +
-    ')' +
-    '\\s*\\))',
-  transform =
-    '(?:' +
-    matrix +
-    '|' +
-    translate +
-    '|' +
-    scale +
-    '|' +
-    rotate +
-    '|' +
-    skewX +
-    '|' +
-    skewY +
-    ')',
-  transforms =
-    '(?:' + transform + '(?:' + commaWsp + '*' + transform + ')*' + ')',
-  transformList = '^\\s*(?:' + transforms + '?)\\s*$',
-  // http://www.w3.org/TR/SVG/coords.html#TransformAttribute
-  reTransformList = new RegExp(transformList),
-  // == end transform regexp
-  reTransform = new RegExp(transform, 'g');
+const p = `(${reNum})`;
+const skewX = String.raw`(skewX)\(${p}\)`;
+const skewY = String.raw`(skewY)\(${p}\)`;
+const rotate = String.raw`(rotate)\(${p}(?: ${p} ${p})?\)`;
+const scale = String.raw`(scale)\(${p}(?: ${p})?\)`;
+const translate = String.raw`(translate)\(${p}(?: ${p})?\)`;
+const matrix = String.raw`(matrix)\(${p} ${p} ${p} ${p} ${p} ${p}\)`;
+const transform = `(?:${matrix}|${translate}|${rotate}|${scale}|${skewX}|${skewY})`;
+const transforms = `(?:${transform}*)`;
+const transformList = String.raw`^\s*(?:${transforms}?)\s*$`;
+// http://www.w3.org/TR/SVG/coords.html#TransformAttribute
+const reTransformList = new RegExp(transformList);
+// == end transform regexp
+const reTransform = new RegExp(transform, 'g');
 
 /**
  * Parses "transform" attribute, returning an array of values
  * @static
  * @function
  * @memberOf fabric
  * @param {String} attributeValue String containing attribute value
- * @return {Array} Array of 6 elements representing transformation matrix
+ * @return {TTransformMatrix} Array of 6 elements representing transformation matrix
  */
-export function parseTransformAttribute(attributeValue) {
+export function parseTransformAttribute(attributeValue: string): TMat2D {
+  // first we clean the string
+  attributeValue = attributeValue
+    .replace(new RegExp(`(${reNum})`, 'gi'), ' $1 ')
+    // replace annoying commas and arbitrary whitespace with single spaces
+    .replace(/,/gi, ' ')
+    .replace(/\s+/gi, ' ')
```

Comment:
actually, sorry, this part here is duplicated in path. Do we want to do a function util, under utils/internals that clean up the string? so we can also test that the input and output of those 3 asserts combined are always what we want.

Reply:
it can totally be called `removeExtraSpaceAndCommasFromSVGAttributes` it doesn't have to be generic or anything like that.

Reply:
If it is used more than once, sure

Reply:
Even if used once

Reply:
i don't know, it could be because we replace first the comma with a 'space' and only then we can clean up multiple spaces. Probably this is safer to follow and a single regexp is a bit more hard to read and be sure it does what it has to do.